### PR TITLE
Allow hosts override from extra vars

### PR DESCRIPTION
**Pull request by jpodivin**
Sometimes we may want to apply our play to a subset of nodes, or to all nodes, even though it was originally targeting 'all' or just some of them.

Right now, all of our playbooks target all hosts by default, so the override can be hypothetically used to target subset of nodes. But eventually, when we have playbooks for specific groups in inventory, this will allow us to instead go for all of them.

We just need to document it.

### CODE REVIEW

**Review by rebtoor — COMMENTED** [review]
My +2 on this, i only left a comment for a typo in the doc.

Does it worth to add a `, true` after the default value in each hosts definition to fallback to default even if `edpm_override_hosts` is defined but empty?

**Comment by jpodivin** [comment]
> My +2 on this, i only left a comment for a typo in the doc.
> 
> Does it worth to add a `, true` after the default value in each hosts definition to fallback to default even if `edpm_override_hosts` is defined but empty?

That is a good point. I'll add that right away.

**Comment by jpodivin** [comment]
This has to wait until we merge crun regression mitigation reverts.

**Comment by jpodivin** [comment]
recheck - crun regression workaround removed

File: docs/source/contributing_playbooks.rst
Comment (slagle) [inline docs/source/contributing_playbooks.rst:26]: must not?

File: docs/source/contributing_playbooks.rst
Comment (slagle) [inline docs/source/contributing_playbooks.rst:22]: I don't really agree with the analogy. Could we link to ansible docs describing playbooks instead of trying to do it ourselves? It seems redundant. I'd just link to https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_intro.html

File: docs/source/contributing_playbooks.rst
Comment (slagle) [inline docs/source/contributing_playbooks.rst:66]: why would we allow any license other than the repo license? I don't feel like this statement is needed. Perhaps general contributing guidance should say something like:

All content contributed to edpm-ansible must be licensed the same as edpm-ansible (`Apache v2.0`).

File: docs/source/contributing_playbooks.rst
Comment (slagle) [inline docs/source/contributing_playbooks.rst:64]: I don't think "Beware the link rot" is a useful idiom in our official docs

File: docs/source/contributing_playbooks.rst
Comment (slagle) [inline docs/source/contributing_playbooks.rst:87]: `A new` or `Each new`

File: docs/source/contributing_playbooks.rst
Comment (slagle) [inline docs/source/contributing_playbooks.rst:21]: Generally, yes. But for the purposes of edpm-ansible I'd prefer that all tasks are in roles and playbooks only `import_role`. A quick check of all existing playbooks show this to be the case. Could we make this a rule, and only consider exceptions as needed?

**Review by slagle — COMMENTED** [review]
change looks good, i have several comments on the playbook docs

File: docs/source/contributing_playbooks.rst
Comment (jpodivin) [reply docs/source/contributing_playbooks.rst:26]: yep.

File: docs/source/contributing_playbooks.rst
Comment (jpodivin) [reply docs/source/contributing_playbooks.rst:66]: In general we could just say "use apache" but there are compatible licenses and I recall we had some cases in tripleo when we had them rather than then the primary. I suppose it's not really likely to be a deal breaker for anyone so I'll just remove the sentence.

File: docs/source/contributing_playbooks.rst
Comment (jpodivin) [reply docs/source/contributing_playbooks.rst:21]: I suppose that is a valid approach.

File: docs/source/contributing_playbooks.rst
Comment (slagle) [inline docs/source/contributing_playbooks.rst:26]: through

File: docs/source/contributing_playbooks.rst
Comment (slagle) [inline docs/source/contributing_playbooks.rst:34]: `default('all', true)`

**Review by slagle — APPROVED** [review]
LGTM.
please fix the couple of inline comments in a follow up.

**Pull request merged**
